File: JavaScript/Basics/0.6-Nested-Loops/Exercises/special-numbers.js
function specialNumber(input) {
    let n = Number(input[0]);
    let specialNumbers = [];
    for (let num = 1111; num <= 9999; num++) {
        num = `${num}`; isSpecial = true;
        for (i = 0; i < num.length; i++) {
            let x = Number(num[i]);
            if (!(n % x == 0)) {isSpecial = false; break;}
        }
        if (isSpecial) {specialNumbers.push(num);}
    }
    return specialNumbers.join(' ');
}


console.log(specialNumber(["3"]));
console.log(specialNumber(["11"]));
console.log(specialNumber(["16"]));

File: Programing-Fundamentals-With-Python/0-2-Data-Types-And-Variables/Lab/special_numbers.py
number = int(input())
special_numbers = [5, 7, 11]

for n in range(1, number + 1):
    sum_of_digits = sum(int(d) for d in str(n))
    is_special = False
    if sum_of_digits in special_numbers:
        is_special = True

    print(f"{n} -> {is_special}")
    